improver_tests/ensemble_copula_coupling/test_utilities.py

```diff
@@ -398,5 +402,59 @@ def test_multiple_timesteps(self):
         self.assertArrayAlmostEqual(reshaped_array, expected)
 
 
+numba_installed = True
+try:
+    importlib.util.find_spec("numba")
+    from improver.ensemble_copula_coupling.numba_utilities import fast_interp_same_y
+except ImportError:
+    numba_installed = False
+
+
+class TestInterpolateMultipleRows(IrisTest):
+
+    """Test interpolate_multiple_rows"""
+
+    def setUp(self):
+        """Set up arrays."""
+        np.random.seed(0)
+        self.x = np.arange(0, 1, 0.01)
+        self.xp = np.sort(np.random.random_sample((100, 100)), axis=1)
+        self.fp = np.arange(0, 100, 1).astype(float)
+
+    @skipIf(not (numba_installed), "numba not installed")
+    def test_slow_vs_fast(self):
+        """Test that slow and fast versions give same result."""
+        result_slow = slow_interp_same_y(self.x, self.xp, self.fp)
+        result_fast = fast_interp_same_y(self.x, self.xp, self.fp)
+        np.testing.assert_allclose(result_slow, result_fast)
+
+    @skipIf(not (numba_installed), "numba not installed")
+    def test_slow_vs_fast_unordered(self):
+        """Test that slow and fast versions give same result
+        when x is not sorted."""
+        shuffled_x = self.x.copy()
+        np.random.shuffle(shuffled_x)
+        result_slow = slow_interp_same_y(shuffled_x, self.xp, self.fp)
+        result_fast = fast_interp_same_y(shuffled_x, self.xp, self.fp)
+        np.testing.assert_allclose(result_slow, result_fast)
+
+    @skipIf(not (numba_installed), "numba not installed")
+    def test_slow_vs_fast_repeated(self):
+        """Test that slow and fast versions give same result when
+        rows of xp contain repeats."""
+        xp_repeat = self.xp.copy()
+        xp_repeat[:, 51] = xp_repeat[:, 50]
+        result_slow = slow_interp_same_y(self.x, xp_repeat, self.fp)
+        result_fast = fast_interp_same_y(self.x, xp_repeat, self.fp)
+        np.testing.assert_allclose(result_slow, result_fast)
+
+    def test_slow_vs_multi(self):
```

Comment:
This test only compares with itself in the absence of numba, so may as well be excluded if numba is not present.